libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -195,16 +184,17 @@ public void subscribe(LifecycleOwner owner, NavigationViewModel navigationViewMo
     lifecycleOwner.getLifecycle().addObserver(this);
     this.navigationViewModel = navigationViewModel;
 
+    navigationViewModel.retrieveBannerInstructions().observe(lifecycleOwner, bannerInstructions -> {
+      toggleGuidanceView(bannerInstructions);
```

Comment:
do we need to expose this as a public api?
I am just curious can this logic be handled inside the InstructionView or maybe we could expose an option to let user turn on/off `GuidanceView`.
So we won't introduce another `SEMVER` change in near future to move the `GuidanceView` out of `InstructionView`.

Reply:
None of our customers use `InstructionView`. I will let it be as is for now, because I don't see why we would want to move it out as a separate view. All it is an image view. Customers have been drawing guidance views by themselves as they are available in banner instructions.

Reply:
since we talked about this several days ago when we discussed the GuidanceView landscape UX and we were talking about moving the guidanceView logic out of InstructionView.
It's fine that not having an individual component just for the guidance view as you mentioned above it's just an image view.

But I think exposing this new public API in InstructionView seems weird. We may leave it here for now if you prefer. But we should think of having a separate GuidanceImageObserver, or sort of, in Core and the developer who wants guidanceImage can just register for that without knowing the InstructionView.
This way, we can solve the 1tap guidance view problem(https://github.com/mapbox/mapbox-navigation-android/issues/3303) too. As you said, 1tap can draw the image wherever they like. cc: @LukasPaczos

Reply:
Yes, and all 1TAP needs to do is the following without any dependency from UI SDK. Inside their `BannerInstructionObserver` 
```
val bannerView = bannerInstruction.view();
List<BannerComponents> components = bannerView.components();
    if (components != null) {
      String urlToLoad = components.get(0).imageUrl();
      new Picasso
          .Builder(getContext())
          .downloader(new OkHttp3Downloader(getClient()))
          .build()
          .load(urlToLoad)
          .into(guidanceViewImage);
    }
```
This way you don't need to use the GuidanceView from `InstructionView`, but if you are happy with Mapbox's implementation, you can just invoke the API as provided.

Reply:
> This way you don't need to use the GuidanceView from InstructionView, but if you are happy with Mapbox's implementation, you can just invoke the API as provided.

Yea, I think this is flexible enough, we shouldn't need another observer just for the guidance view type :+1: 

There is one caveat though, the user agent has to contain `MapboxJava`. We should document it in the `BannerInstruction` object for any developers that would like to make the request themselves. Unless this is changing soon @danpaz?